make_release.py

```diff
@@ -142,149 +163,185 @@ def short(rel_name):
+    '0.8.0'
+    >>> long_release_name('0.8rc1')
+    '0.8.0rc1'
+    >>> long_release_name('0.8.0rc1')
+    '0.8.0rc1'
+    """
+    dotcount = release_name.count('.')
+    if dotcount >= 2:
+        return release_name
+    assert dotcount == 1, "{} contains {} dots".format(release_name, dotcount)
+    pos = pretag_pos(release_name)
+    if pos is not None:
+        return release_name[:pos] + '.0' + release_name[pos:]
+    return release_name + '.0'
+
+
+def pretag_pos(release_name):
+    """
+    gives the position of any pre-release tag
+    >>> pretag_pos('0.8')
+    >>> pretag_pos('0.8alpha25')
+    3
+    >>> pretag_pos('0.8.1rc1')
+    5
+    """
+    # 'a' needs to be searched for after 'beta'
+    for tag in ('rc', 'c', 'beta', 'b', 'alpha', 'a'):
+        match = re.search(tag + '\d+', release_name)
+        if match is not None:
+            return match.start()
+    return None
+
+
 def strip_pretags(release_name):
     """
     removes pre-release tags from a version string
-
-    >>> str(strip_pretags('0.8'))
+    >>> strip_pretags('0.8')
     '0.8'
-    >>> str(strip_pretags('0.8alpha25'))
+    >>> strip_pretags('0.8alpha25')
     '0.8'
-    >>> str(strip_pretags('0.8.1rc1'))
+    >>> strip_pretags('0.8.1rc1')
     '0.8.1'
     """
-    # 'a' needs to be searched for after 'beta'
-    for tag in ('rc', 'c', 'beta', 'b', 'alpha', 'a'):
-        release_name = re.sub(tag + '\d+', '', release_name)
-    return release_name
+    pos = pretag_pos(release_name)
+    return release_name[:pos] if pos is not None else release_name
 
 
 def isprerelease(release_name):
     """
     tests whether the release name contains any pre-release tag
-
     >>> isprerelease('0.8')
     False
     >>> isprerelease('0.8alpha25')
     True
     >>> isprerelease('0.8.1rc1')
     True
     """
-    return any(tag in release_name for tag in ('rc', 'c', 'beta', 'b', 'alpha', 'a'))
+    return pretag_pos(release_name) is not None
 
 
 # -------------------- #
 # end of generic tools #
 # -------------------- #
 
+# ------------------------- #
+# specific helper functions #
+# ------------------------- #
+
+
 def relname2fname(release_name):
     short_version = short(strip_pretags(release_name))
-    return r"version_%s.rst.inc" % short_version.replace('.', '_')
+    return r"version_{}.rst.inc".format(short_version.replace('.', '_'))
 
 
-def release_changes(release_name):
-    fpath = r"doc\source\changes\\" + relname2fname(release_name)
-    with open(fpath, encoding='utf-8-sig') as f:
+def release_changes(context):
+    directory = r"doc\source\changes"
+    fname = relname2fname(context['release_name'])
+    with open(os.path.join(context['build_dir'], directory, fname), encoding='utf-8-sig') as f:
         return f.read()
 
 
-def build_doc():
-    chdir('doc')
-    call('buildall.bat')
-    chdir('..')
-
-
-def update_changelog(release_name):
-    fname = relname2fname(release_name)
+def replace_lines(fpath, changes, end="\n"):
```

Comment:
Why have an end argument? Is there a case where it is needed?

Reply:
Not now but it might.